Tests/Marketplace/upload_packs.py

```diff
@@ -169,6 +171,56 @@ def update_index_folder(index_folder_path, pack_name, pack_path, pack_version=''
         return task_status
 
 
+def clean_non_existing_packs(index_folder_path, private_packs, storage_bucket):
+    """ Detects packs that are not part of content repo or from private packs bucket.
+
+    In case such packs were detected, problematic pack is deleted from index and from content/packs/{target_pack} path.
+
+    Args:
+        index_folder_path (str): full path to downloaded index folder.
+        private_packs (list): priced packs from private bucket.
+        storage_bucket (google.cloud.storage.bucket.Bucket): google storage bucket where index.zip is stored.
+
+    Returns:
+        bool: whether cleanup was skipped or not.
+    """
+    if 'CI' not in os.environ or os.environ.get('CIRCLE_BRANCH') != 'master' \
+            or storage_bucket.name != GCPConfig.PRODUCTION_BUCKET:
```

Comment:
in case we are NOT on the PRODUCTION_BUCKET, I think we should run the cleanup. this way wi will cleanup the build bucket and it will be a better representation of whats going to happen if a branch will be merged.

Reply:
@yuvalbenshalom hmm I am not sure about non production bucket. The ci bucket is copied from master/production bucket (all master branches will have cleanup), so I don't see any case where it will ever enter the cleanup part on non production bucket.